sunpy/net/scraper.py

```diff
@@ -72,24 +102,55 @@ class Scraper:
     pattern looks with the actual time.
     """
 
-    def __init__(self, pattern, **kwargs):
-        pattern = pattern.format(**kwargs)
-        timepattern = pattern
-        for k, v in TIME_CONVERSIONS.items():
-            if k in timepattern:
-                timepattern = timepattern.replace(k,v)
-        self.timepattern = timepattern
-        if "year:4d" in pattern and "year:2d" in pattern:
-            pattern = pattern.replace("year:2d", ":2d")
-        self.pattern = pattern
-        self.domain = f"{urlsplit(self.pattern).scheme}://{urlsplit(self.pattern).netloc}/"
-        milliseconds = re.search(r'\%e', self.timepattern)
-        if not milliseconds:
-            self.now = datetime.now().strftime(self.timepattern)
+    def __init__(self, pattern=None, regex=False, format=None, **kwargs):
+        if pattern is not None and format is None:
+            warnings.warn(
+                "Using 'pattern' for the URL format is deprecated. "
+                "Please use 'format' instead.",
+                DeprecationWarning,
```

Comment:
This should be a sunpydepcreationwaring instead.

Reply:
hey i can't seem to figure out which of the existing decorators would apply in this case, [here they are for reference](https://github.com/sunpy/sunpy/blob/main/sunpy/util/decorators.py). does one of them fit here or does this call for a new one for deprecating specific arguments?